### PR TITLE
[pluggable monitor] add `--silent` and `--config` flags

### PR DESCRIPTION
**Pull request by cmaglie**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

* **What kind of change does this PR introduce?**
This PR adds two flags to the `monitor` command of the CLI:
1. `--config/-c` allows to set configuration of the monitor port before opening for example:
```
$ ./arduino-cli monitor -p /dev/ttyACM0 -c 115200,None,8,1
Set baudrate to 115200
Set parity to None
Set bits to 8
Set stop_bits to 1
Connected to /dev/ttyACM0! Press CTRL-C to exit.
HELLO 33833
HELLO 33834
HELLO 33835
HELLO 33836
HELLO 33837
HELLO 33838
HELLO 33839
HELLO 33840
```
The available parameters and values can be obtained with the `--describe` flag.
If an option is ambiguous it can be explicitly set using the configuration setting id followed by `=` and the desired setting value for example:
```
$ ./arduino-cli monitor -p /dev/ttyACM0 --config baudrate=115200,stop_bits=1
```

2. `--silent/-s` will set the CLI in silent mode and will print only the output incoming from the port. This option is enabled automatically when the output of the CLI is redirected to a file but it can be forced with the `-s` flag if the autodetection do not work for some reason.
```
$ ./arduino-cli monitor -p /dev/ttyACM0 -c 9600,None,8,1 -s
HELLO 35574
HELLO 35575
HELLO 35576
HELLO 35577
HELLO 35578
^C
$ ./arduino-cli monitor -p /dev/ttyACM0 -c 9600,None,8,1 > test.log
^C
$ cat test.log 
HELLO 35615
HELLO 35616
HELLO 35617
HELLO 35618
HELLO 35619
HELLO 35620
HELLO 35621
HELLO 35622
HELLO 35623
```

### CODE REVIEW

File: cli/monitor/monitor.go [hunk 1]
```diff
@@ -87,10 +97,63 @@ func runMonitorCmd(cmd *cobra.Command, args []string) {
...
+			if setting == nil {
+				feedback.Error(tr("invalid port configuration: %s", config))
+				os.Exit(errorcodes.ErrBadArgument)
+			}
+			configuration.Settings = append(configuration.Settings, &rpc.MonitorPortSetting{
+				SettingId: setting.SettingId,
+				Value:     v,
+			})
+			if !silent {
+				feedback.Print(fmt.Sprintf("Set %s to %s", setting.SettingId, v))
```
Comment (silvanocerza) [inline cli/monitor/monitor.go:148]:
```suggestion
				feedback.Print(tr("Set %s to %s", setting.SettingId, v))
```

**Review by silvanocerza — CHANGES_REQUESTED** [review]
Some small changes to make.

File: cli/monitor/monitor.go [hunk 2]
```diff
@@ -87,10 +97,63 @@ func runMonitorCmd(cmd *cobra.Command, args []string) {
...
+				} else {
+					if strings.EqualFold(s.SettingId, k) {
+						if !contains(s.EnumValues, v) {
+							feedback.Error(tr("invalid port configuration value for %s: %s", k, v))
+							os.Exit(errorcodes.ErrBadArgument)
+						}
+						setting = s
+						break
+					}
+				}
```
Comment (silvanocerza) [inline cli/monitor/monitor.go:137]:
```suggestion
				if k == "" && contains(s.EnumValues, v){
					setting = s
					break
				} else if strings.EqualFold(s.SettingId, k) {
					if !contains(s.EnumValues, v) {
						feedback.Error(tr("invalid port configuration value for %s: %s", k, v))
						os.Exit(errorcodes.ErrBadArgument)
					}
					setting = s
					break
				}
```

File: cli/monitor/monitor.go (hunk 1 above)
Comment (cmaglie) [reply cli/monitor/monitor.go]: Since the original `Set %s to %s` doesn't give enough context for translation I changed to `Monitor port settings:`

**Review by silvanocerza — APPROVED** [review]

**Pull request merged**